src/tools/solver_protocol.h

```diff
@@ -0,0 +1,13 @@
+#ifndef EPIC_SOLVER_PROTOCOL_H
+#define EPIC_SOLVER_PROTOCOL_H
+
+namespace SolverResult{
+    enum ErrorCode{
```

Comment:
if we won't continue using this namespace, I prefer 
```suggestion
    enum class ErrorCode{
```